TestSwitchApi/TestSwitchApi/Services/SessionService.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using Microsoft.AspNetCore.Http;
+using TestSwitchApi.Repositories;
+
+namespace TestSwitchApi.Services
+{
+    public class SessionService : ISessionService
+    {
+        public bool SessionInDate(DateTime SessionEndTime)
+        {
+            return SessionEndTime > DateTime.Now ? true : false;
+        }
+
+        public bool RequestHasValidSessionId(HttpContext context, IAdminRepo adminRepo)
```

Comment:
You can just inject an `IAdminRepo` into this class to make it all a little simpler.
That way you don't need to pass in this dependency.

Reply:
does that work the same way as the controller?

Reply:
yep - just add it into the constructor and it should just work